content/en/blog/2023/tyk-api-gateway/index.md

```diff
@@ -0,0 +1,61 @@
+
+### Enable OpenTelemetry at Gateway Level
+
+Edit Tyk Gateway configuration file to enable OpenTelemetry support, like this:
+
+{
+  "opentelemetry": {
+    "enabled": true
+  }
+}
+
+You can also enable OpenTelemetry by setting the corresponding environment variable: TYK_GW_OPENTELEMETRY_ENABLED=TRUE.
+
+By default, OpenTelemetry spans are exported using the gRPC protocol to localhost:4317. For more configuration options and default values, refer to the [documentation](https://tyk.io/docs/tyk-oss-gateway/configuration/#opentelemetry).
+
+Tyk will export two spans: one for the total duration of the API call and one span for the call to the upstream service, as shown in the following screenshot from Jaeger. 
+
+![Typ API Gateway distributed trace](tyk-api-gateway-opentelemetry-trace.png)
+
+
+### Detailed tracing per APIs
+
+Upon enabling OpenTelemetry at the gateway level, you gain the flexibility to activate detailed tracing for particular APIs. To do so, simply adjust the corresponding API definition and toggle the 'detailed_tracing' option to either 'true' or 'false.' The default value for this setting is 'false.'
+With this configuration, Tyk will generate a span for every middleware operation carried out throughout the API request/response lifecycle. This includes actions like API version validation, authentication, authorization, rate limiting, and request and response transformation.
+
+Here you can quickly spot an issue reported by the rate limiting middleware: 
+
+![Typ API Gateway distributed trace with middleware details](tyk-api-gateway-opentelemetry-trace-detail.png)
+
+### Instrumenting custom plugins with OpenTelemetry
+
+Tyk supports the use of custom plugins (also called custom middleware) to extend its functionality, making it a highly flexible and customizable API gateway solution. This extensibility allows developers to tailor Tyk to their specific needs, whether it's for enhancing security, adding new authentication methods, integrating with third-party services, or implementing custom request/response transformations.
+
+By instrumenting custom plugins with OpenTelemetry SDK or Tyk’s OpenTelemetry library implementation (wrapper of the OpenTelemetry SDK that all Tyk components are sharing for convenience), Tyk users can gain additional insights into their custom plugin behaviour for better monitoring and quicker troubleshooting. More information in Tyk’s documentation: [OpenTelemetry Instrumentation in Go Plugins](https://tyk.io/docs/product-stack/tyk-gateway/advanced-configurations/plugins/otel-plugins/). 
+
+## What's next?
+
+Now that the [first stable release of OTel Go metric SDK has been announced](https://opentelemetry.io/blog/2023/otel-go-metrics-sdk-stable/), we will be investigating support for OpenTelemetry metrics. Stay tuned for updates and don't hesitate to reach out with any feedback.
```

Comment:
Provide a link where people can provide feedback (e.g. github issues?)

Reply:
done - have added the link to our community.